d3-series/materials/Week 6/Assignments/us-employment.html

```diff
@@ -0,0 +1,150 @@
+            }
+
+            function prepData(data) {
+                return data
+                    .map(function(d) {
+                        return {
+                            id: d.StateCode + d.CountyCode,
+                            unemployee: +d.Percent
+                        };
+                    });
+            }
+
+            function prepDataUnEmploy(data) {
+                var unemploy = {};
+
+                data.forEach(function(d) {
+                    unemploy[+(d.StateCode + d.CountyCode)] = +d.Percent;
+                   
+                });
+                console.log('unemploy',unemploy[1001]);
+                return unemploy;
+            }
+
+            function draw(geojson, lct, ww, hh) {
+                var width = 250;
+                var height = 200;
+
+                var margin = {
+                    top: 30,
+                    right: 30,
+                    bottom: 30,
+                    left: 30
+                };
+
+                var innerWidth = width - margin.left - margin.right;
+                var innerHeight = height - margin.top - margin.bottom;
+
+             
+                var svg = d3
+                    .select(containerId)
+                    .append('svg')
+                    .attr('height', height)
+                    .attr('width', width);
+
+                var g = svg
+                    .append('g')
+                    .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
+                geojson.features.forEach(function(f) {
+                    var unemploy = lct[f.id];
+                    if (unemploy) {
+                        f.unemployee = unemploy;
+                    }
+                });
+
+                var opacityScale = d3
+                    .scaleLinear()
+                    .domain([0,30])
+                    .range([0, 1]);
+
+                var usaProj = d3
+                    .geoAlbersUsa()
+                    .scale(300)
+                    .translate([ww, hh]);
+
+                var geoPath = d3.geoPath().projection(usaProj);
+
+                g
+                    .selectAll('path')
+                    .data(geojson.features)
+                    .enter()
+                    .append('path')
+                    .attr('d', geoPath)
+                    .style('fill', function(d) {
+                        if (d.unemployee) {
+                            return 'red';
+                        } else {
+                            return 'grey';
+                        }
+                    })
+                    .style('stroke', 'grey')
+                    .style('stroke-width', 0.1)
+                    .attr('fill-opacity', function(d) {
+                        if (d.unemployee) {
+                            return opacityScale(d.unemployee);
+                        } else {
+                            return 1;
+                        }
+                    });
+
+            }

```

Comment:
This looks good! Going forward it's always good to include a title and legend when applicable to complete the data viz.